File: src/main/java/spring/reactive/chapter14/Example14_1.java
package spring.reactive.chapter14;

import lombok.extern.slf4j.Slf4j;
import reactor.core.publisher.Mono;

@Slf4j
public class Example14_1 {
    public static void main(String[] args) {
        Mono.justOrEmpty(null)
                .subscribe(
                        data -> {},
                        error -> {},
                        () -> log.info("# Completed")
                );
    }
}

File: src/main/kotlin/spring/reactive/chapter14/Example14_1.kt
package spring.reactive.chapter14

import mu.KotlinLogging
import reactor.core.publisher.Mono

fun main() {
    val logger = KotlinLogging.logger {}
    Mono.just { null }
        .subscribe(
            { _ -> },
            { _ -> },
            { logger.info { "# onCompleted" } }
        )
}
